### DISCUSSION TITLE
As an editor with a site with many variants, I'd like to set all scheduled publishing date for all the languages at once.

### DISCUSSION THREAD

**Discussion by simonech** [umbraco/Umbraco-CMS#15296] [Features and ideas]
I'm working on a site with 24 variants, and it's a PITA for editors to set the scheduled date for content since they need to set the same date and time 24 times.

![image](https://github.com/umbraco/Umbraco-CMS/assets/61557/7453b7a7-b63c-4abf-a3e5-fcee044585d3)

Would be useful if there was a way to set the same date, for all the available languages at once.

**Comment by lassefredslund** [comment]
Thanks @simonech 
I see that you (and others) could benefit from a simplified way to set the same date for all languages. I have marked this as "up for grabs"

**Reply by simonech** [reply]
Hi @lassefredslund 
We are finally looking into building this feature.
While the ["select all languages" for publishing ](https://github.com/umbraco/Umbraco-CMS/discussions/15296) was more straightforward, this one is a bit more complicated UI-wise.
So, before implementing it, we wanted to check with you (and maybe @nul800sebastiaan) about the proper way to do it.

We will do it in v13 for the moment; hopefully, some smarter front-end developer in HQ can port it to v14.

For the UX implementation, we were thinking of adding 2 UI elements:

- a checkbox to select all created languages
- a button to copy the dates set in the mandatory language to all the selected languages

The idea was to decouple the selection from the copy of the dates so that you can also manually select some languages and still have the copy date feature.

What do you think?

**Reply by lassefredslund** [reply]
Hi @simonech,

Thank you so much for your suggestion.
We've briefly talked about it here at HQ. What do you think about an approach like this (sorry for the very speedy mockups):

Before any fallback dates have been set:
![Mockup1](https://github.com/umbraco/Umbraco-CMS/assets/101121881/acaff8d9-55c9-4d6d-b286-5c7654da4c13)

After a fallback publish date has been set (and with a deselected language):
![Mockup2](https://github.com/umbraco/Umbraco-CMS/assets/101121881/24de9c2e-a777-4852-9457-d41401a92867)

This approach brings the check-all button but also a fallback-publish and fallback-unpublish date field for all variants.
Each variant will have to be checked to be part of the scheduling. If not checked, nothing is done. The Check-all button can be used to select or de-select all. For a selected variant the Publish or Unpublish date will inherit the fallback date unless a specific date is defined in this field.

For the user to understand if it's an inherited or specified date, we should display the fallback date with a placeholder look (grey text).
This means, that if a fallback date is defined then this date is automatically used for all checked variants that do not have a more specific date defined.

The user should be able to define the fallback dates no matter if the ‘all variants’ checkbox is checked or not, meaning that the fallback input fields are always visible.

**Reply by simonech** [reply]
Hi @lassefredslund 
Speedy mockups are better than my "no mockup" 😀
I think I understood what you mean.
Thank you for the time in reviewing the proposal.
Simone

**Comment by AndyButland** [comment]
I've proposed an approach to this in https://github.com/umbraco/Umbraco-CMS/pull/23375.
